### PR TITLE
Update to new ROI commands

### PR DESCRIPTION
**Pull request by DonLakeFlyer**
* Implements new ROI_LOCATION and ROI_NONE using existing code from DO_SET_ROI implementation.
* ROI_WPNEXT_OFFSET is plumbed through. The actual implementation is still NYI.
* Unsure how this will collide with #8620.
* Tested as much as possible which is mostly upload and fly without error. Without #8620 ROI doesn't really work so can't validate fully.

QGC support: https://github.com/mavlink/qgroundcontrol/pull/6017

### CODE REVIEW

File: src/modules/commander/commander.cpp [hunk 1]
```diff
@@ -1162,6 +1162,9 @@ Commander::handle_command(vehicle_status_s *status_local, const safety_s *safety
 	case vehicle_command_s::VEHICLE_CMD_NAV_DELAY:
 	case vehicle_command_s::VEHICLE_CMD_DO_SET_ROI:
 	case vehicle_command_s::VEHICLE_CMD_NAV_ROI:
+    case vehicle_command_s::VEHICLE_CMD_DO_SET_ROI_LOCATION:
```
Comment (ndepal) [inline src/modules/commander/commander.cpp]: Indentation

File: src/modules/navigator/mission_feasibility_checker.cpp [hunk 2]
```diff
@@ -268,6 +268,9 @@ MissionFeasibilityChecker::checkMissionItemValidity(const mission_s &mission)
 		    missionitem.nav_cmd != NAV_CMD_DO_MOUNT_CONFIGURE &&
 		    missionitem.nav_cmd != NAV_CMD_DO_MOUNT_CONTROL &&
 		    missionitem.nav_cmd != NAV_CMD_DO_SET_ROI &&
+			missionitem.nav_cmd != NAV_CMD_DO_SET_ROI_LOCATION &&
+			missionitem.nav_cmd != NAV_CMD_DO_SET_ROI_WPNEXT_OFFSET &&
+			missionitem.nav_cmd != NAV_CMD_DO_SET_ROI_NONE &&
```
Comment (ndepal) [inline src/modules/navigator/mission_feasibility_checker.cpp]: Indentation

**Comment by ndepal** [comment]
> Unsure how this will collide with #8620.

I'll probably have to re-do that PR. But it doesn't achieve the desired result anyway.
Am I correct in understanding that this PR does not introduce any functional changes?

While we're changing message definitions, it would be nice to address the absolute/relative alt problem https://github.com/PX4/Firmware/pull/8595/files/29c90e96646e3bec5484f92e5ad66dd1b9a49e83#diff-7a824e3d1e9a5420071c8c9bbe27b8e7R454

**Comment by dagar** [comment]
Build failures are trivial formatting issues. Locally you can install astyle and run `make format`.

**Comment by DonLakeFlyer** [comment]
> Am I correct in understanding that this PR does not introduce any functional changes?

That is correct. It just connects the new commands to the existing code.

**Comment by DonLakeFlyer** [comment]
I'm not sure I understand what the relative alt problem is?

**Comment by ndepal** [comment]
> I'm not sure I understand what the relative alt problem is?

`VEHICLE_CMD_DO_SET_ROI_LOCATION` has an `altitude` field. It is not clear whether this is altitude above sea or above home (relative altitude).

When specifying a `MAV_CMD_DO_SET_ROI` in a mission, it is possible to specify that the altitude is relative to home. This information used to get lost when translating into a `vehicle_command`.
I worked around this for `MAV_CMD_DO_SET_ROI` commands in missions (see my link above to the relevant lines). This does not work for commands sent over mavlink as `COMMAND_INT/LONG` where the relative/absolute alt information does not exist.

**Comment by DonLakeFlyer** [comment]
Ah, I see. There are quite a few commands which have the same problem. Possible solution:
* Create a COMMAND_LONG_2 which takes an additional MAV_FRAME argument
* Spec out that commands which have alt when sent using COMMAND_LONG are always AMSL

**Comment by dagar** [comment]
Can't you use COMMAND_INT? http://mavlink.org/messages/common#COMMAND_INT
Note the frame.

**Comment by ndepal** [comment]
Thanks for pointing that out @dagar 
I guess I don't really understand the difference between the various `COMMAND_*` messages. I had just assumed they just had different data types (suitable for different links?), but the information conveyed should be the same.

If that's not the case, I suppose we could just only support `COMMAND_INT` for `MAV_CMD_DO_SET_ROI_LOCATION`.

**Comment by DonLakeFlyer** [comment]
> Can't you use COMMAND_INT?

That's a strange command I've never seen before. Seems like an old dup of MISSION_ITEM_INT? It's not really for commands per se since it includes things like current and autocontinue which is mission stuff.

**Comment by LorenzMeier** [comment]
@DonLakeFlyer That's the new command format which is required to be able to send Geo coordinates without distortion. If QGC is not using it it actually means that GOTO is broken because the maximum resolution is 1.4 meters (due to floating point resolution limits).

**Comment by DonLakeFlyer** [comment]
And it contains current and autocontinue why?

**Comment by LorenzMeier** [comment]
The idea back then was to enable these commands to be embedded into a command state machine as async commands. If you set "auto continue" it would then continue with the next in the mission if one was active. That would allow you to send a change altitude command and continue with the next mission item. @dagar FYI. It's not really built out, so I'm not sure if the reasoning makes still sense today. However, this is how it is now in the protocol.

**Comment by DonLakeFlyer** [comment]
@LorenzMeier @ndepal 

I just testing this pull rebased to current master. The plan just does and ROI_LOCATION for a couple waypoints and then an ROI_NONE.
[ROITesting.plan.zip](https://github.com/PX4/Firmware/files/1653234/ROITesting.plan.zip)

MIS_MNT_YAW_CTL = 0
* Correct: Vehicle yawed correctly to point to ROI_LOCATION at correct mission item
* Incorrect: At ROI_LOCATION item, gimbal pointed straight down (or close to that)
* Correct: Vehicle went back to pointing along waypoint path at ROI_NONE

MIS_MNT_YAW_CTL = 1
* Correct: Vehicle did not yaw at ROI_LOCATION item
* Incorrect: Gimbal pitched down at what looked like 45 degrees, did not yaw at ROI_LOCATION
* Correct: Vehicle went back to pointing along waypoint path at ROI_NONE

**Comment by LorenzMeier** [comment]
Could you rebase? Should I rebase?

**Comment by DonLakeFlyer** [comment]
I just pushed up the rebase I had locally.

**Comment by DonLakeFlyer** [comment]
And of course the reasbe went bonkers. Let me resubmit.

**Pull request closed without merging**